In [3]:
import pandas as pd
import numpy as np
import warnings

warnings.filterwarnings('ignore')

In [4]:
data = pd.read_csv("/Users/aytackolukisa/Desktop/Jupiter_Dosyaları/states_all_extended.csv")

In [5]:
data.head(5)

PRIMARY_KEY       STATE  YEAR  ENROLL  TOTAL_REVENUE  FEDERAL_REVENUE  \
0     1992_ALABAMA     ALABAMA  1992     NaN      2678885.0         304177.0   
1      1992_ALASKA      ALASKA  1992     NaN      1049591.0         106780.0   
2     1992_ARIZONA     ARIZONA  1992     NaN      3258079.0         297888.0   
3    1992_ARKANSAS    ARKANSAS  1992     NaN      1711959.0         178571.0   
4  1992_CALIFORNIA  CALIFORNIA  1992     NaN     26260025.0        2072470.0   

   STATE_REVENUE  LOCAL_REVENUE  TOTAL_EXPENDITURE  INSTRUCTION_EXPENDITURE  \
0      1659028.0       715680.0          2653798.0                1481703.0   
1       720711.0       222100.0           972488.0                 498362.0   
2      1369815.0      1590376.0          3401580.0                1435908.0   
3       958785.0       574603.0          1743022.0                 964323.0   
4     16546514.0      7641041.0         27138832.0               14358922.0   

   ...  G08_HI_A_READING  G08_HI_A_MATHEMATICS  G08_AS_A_READING  \
0  ...               NaN                   NaN               NaN   
1  ...               NaN                   NaN               NaN   
2  ...               NaN                   NaN               NaN   
3  ...               NaN                   NaN               NaN   
4  ...               NaN                   NaN               NaN   

   G08_AS_A_MATHEMATICS  G08_AM_A_READING  G08_AM_A_MATHEMATICS  \
0                   NaN               NaN                   NaN   
1                   NaN               NaN                   NaN   
2                   NaN               NaN                   NaN   
3                   NaN               NaN                   NaN   
4                   NaN               NaN                   NaN   

   G08_HP_A_READING  G08_HP_A_MATHEMATICS  G08_TR_A_READING  \
0               NaN                   NaN               NaN   
1               NaN                   NaN               NaN   
2               NaN                   NaN               NaN   
3               NaN                   NaN               NaN   
4               NaN                   NaN               NaN   

   G08_TR_A_MATHEMATICS  
0                   NaN  
1                   NaN  
2                   NaN  
3                   NaN  
4                   NaN  

[5 rows x 266 columns]

In [6]:
data.tail(5)

PRIMARY_KEY          STATE  YEAR  ENROLL  TOTAL_REVENUE  \
1710       2019_VIRGINIA       VIRGINIA  2019     NaN            NaN   
1711     2019_WASHINGTON     WASHINGTON  2019     NaN            NaN   
1712  2019_WEST_VIRGINIA  WEST_VIRGINIA  2019     NaN            NaN   
1713      2019_WISCONSIN      WISCONSIN  2019     NaN            NaN   
1714        2019_WYOMING        WYOMING  2019     NaN            NaN   

      FEDERAL_REVENUE  STATE_REVENUE  LOCAL_REVENUE  TOTAL_EXPENDITURE  \
1710              NaN            NaN            NaN                NaN   
1711              NaN            NaN            NaN                NaN   
1712              NaN            NaN            NaN                NaN   
1713              NaN            NaN            NaN                NaN   
1714              NaN            NaN            NaN                NaN   

      INSTRUCTION_EXPENDITURE  ...  G08_HI_A_READING  G08_HI_A_MATHEMATICS  \
1710                      NaN  ...             247.0                 278.0   
1711                      NaN  ...             248.0                 267.0   
1712                      NaN  ...               NaN                   NaN   
1713                      NaN  ...             251.0                 273.0   
1714                      NaN  ...             254.0                 274.0   

      G08_AS_A_READING  G08_AS_A_MATHEMATICS  G08_AM_A_READING  \
1710             286.0                 315.0               NaN   
1711             285.0                 315.0             237.0   
1712               NaN                   NaN               NaN   
1713             277.0                 294.0             253.0   
1714               NaN                   NaN             243.0   

      G08_AM_A_MATHEMATICS  G08_HP_A_READING  G08_HP_A_MATHEMATICS  \
1710                   NaN               NaN                   NaN   
1711                 259.0               NaN                   NaN   
1712                   NaN               NaN                   NaN   
1713                 267.0               NaN                   NaN   
1714                 258.0               NaN                   NaN   

      G08_TR_A_READING  G08_TR_A_MATHEMATICS  
1710             269.0                 293.0  
1711             263.0                 292.0  
1712             249.0                   NaN  
1713             268.0                 276.0  
1714               NaN                 274.0  

[5 rows x 266 columns]

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1715 entries, 0 to 1714
Columns: 266 entries, PRIMARY_KEY to G08_TR_A_MATHEMATICS
dtypes: float64(263), int64(1), object(2)
memory usage: 3.5+ MB


## Veri Türleri:
    Katagorik_Değişkenler : PRIMARY_KEY,STATE (2 adet)
    Sürekli_Değikenler(int) : YEAR (1 adet)
    Sürekli_Değikenler(float) : TOTAL_REVENUE,FEDERAL_REVENUE,.....G08_TR_A_MATHEMATICS (263 adet)

In [8]:
data.isnull().sum()*100/len(data)

PRIMARY_KEY              0.000000
STATE                    0.000000
YEAR                     0.000000
ENROLL                  28.629738
TOTAL_REVENUE           25.655977
                          ...    
G08_AM_A_MATHEMATICS    96.501458
G08_HP_A_READING        99.183673
G08_HP_A_MATHEMATICS    99.241983
G08_TR_A_READING        91.778426
G08_TR_A_MATHEMATICS    91.545190
Length: 266, dtype: float64

youtube["video_sayisi"].fillna(youtube["video_sayisi"].mean(), inplace=True)


In [9]:
df = data.drop(["PRIMARY_KEY","STATE"],axis=1)

In [10]:
df

YEAR  ENROLL  TOTAL_REVENUE  FEDERAL_REVENUE  STATE_REVENUE  \
0     1992     NaN      2678885.0         304177.0      1659028.0   
1     1992     NaN      1049591.0         106780.0       720711.0   
2     1992     NaN      3258079.0         297888.0      1369815.0   
3     1992     NaN      1711959.0         178571.0       958785.0   
4     1992     NaN     26260025.0        2072470.0     16546514.0   
...    ...     ...            ...              ...            ...   
1710  2019     NaN            NaN              NaN            NaN   
1711  2019     NaN            NaN              NaN            NaN   
1712  2019     NaN            NaN              NaN            NaN   
1713  2019     NaN            NaN              NaN            NaN   
1714  2019     NaN            NaN              NaN            NaN   

      LOCAL_REVENUE  TOTAL_EXPENDITURE  INSTRUCTION_EXPENDITURE  \
0          715680.0          2653798.0                1481703.0   
1          222100.0           972488.0                 498362.0   
2         1590376.0          3401580.0                1435908.0   
3          574603.0          1743022.0                 964323.0   
4         7641041.0         27138832.0               14358922.0   
...             ...                ...                      ...   
1710            NaN                NaN                      NaN   
1711            NaN                NaN                      NaN   
1712            NaN                NaN                      NaN   
1713            NaN                NaN                      NaN   
1714            NaN                NaN                      NaN   

      SUPPORT_SERVICES_EXPENDITURE  OTHER_EXPENDITURE  ...  G08_HI_A_READING  \
0                         735036.0                NaN  ...               NaN   
1                         350902.0                NaN  ...               NaN   
2                        1007732.0                NaN  ...               NaN   
3                         483488.0                NaN  ...               NaN   
4                        8520926.0                NaN  ...               NaN   
...                            ...                ...  ...               ...   
1710                           NaN                NaN  ...             247.0   
1711                           NaN                NaN  ...             248.0   
1712                           NaN                NaN  ...               NaN   
1713                           NaN                NaN  ...             251.0   
1714                           NaN                NaN  ...             254.0   

      G08_HI_A_MATHEMATICS  G08_AS_A_READING  G08_AS_A_MATHEMATICS  \
0                      NaN               NaN                   NaN   
1                      NaN               NaN                   NaN   
2                      NaN               NaN                   NaN   
3                      NaN               NaN                   NaN   
4                      NaN               NaN                   NaN   
...                    ...               ...                   ...   
1710                 278.0             286.0                 315.0   
1711                 267.0             285.0                 315.0   
1712                   NaN               NaN                   NaN   
1713                 273.0             277.0                 294.0   
1714                 274.0               NaN                   NaN   

      G08_AM_A_READING  G08_AM_A_MATHEMATICS  G08_HP_A_READING  \
0                  NaN                   NaN               NaN   
1                  NaN                   NaN               NaN   
2                  NaN                   NaN               NaN   
3                  NaN                   NaN               NaN   
4                  NaN                   NaN               NaN   
...                ...                   ...               ...   
1710               NaN                   NaN               NaN   
1711             237.0                 259

In [11]:
def filling_by_mean(x):
    df[x].fillna(df[x].mean(), inplace=True)
    
    

In [12]:
liste = df.columns

In [13]:
for c in liste:
    filling_by_mean(c)
    

In [14]:
df

YEAR         ENROLL  TOTAL_REVENUE  FEDERAL_REVENUE  STATE_REVENUE  \
0     1992  917541.566176   2.678885e+06     3.041770e+05   1.659028e+06   
1     1992  917541.566176   1.049591e+06     1.067800e+05   7.207110e+05   
2     1992  917541.566176   3.258079e+06     2.978880e+05   1.369815e+06   
3     1992  917541.566176   1.711959e+06     1.785710e+05   9.587850e+05   
4     1992  917541.566176   2.626002e+07     2.072470e+06   1.654651e+07   
...    ...            ...            ...              ...            ...   
1710  2019  917541.566176   9.102045e+06     7.677799e+05   4.223743e+06   
1711  2019  917541.566176   9.102045e+06     7.677799e+05   4.223743e+06   
1712  2019  917541.566176   9.102045e+06     7.677799e+05   4.223743e+06   
1713  2019  917541.566176   9.102045e+06     7.677799e+05   4.223743e+06   
1714  2019  917541.566176   9.102045e+06     7.677799e+05   4.223743e+06   

      LOCAL_REVENUE  TOTAL_EXPENDITURE  INSTRUCTION_EXPENDITURE  \
0      7.156800e+05       2.653798e+06             1.481703e+06   
1      2.221000e+05       9.724880e+05             4.983620e+05   
2      1.590376e+06       3.401580e+06             1.435908e+06   
3      5.746030e+05       1.743022e+06             9.643230e+05   
4      7.641041e+06       2.713883e+07             1.435892e+07   
...             ...                ...                      ...   
1710   4.110522e+06       9.206242e+06             4.768010e+06   
1711   4.110522e+06       9.206242e+06             4.768010e+06   
1712   4.110522e+06       9.206242e+06             4.768010e+06   
1713   4.110522e+06       9.206242e+06             4.768010e+06   
1714   4.110522e+06       9.206242e+06             4.768010e+06   

      SUPPORT_SERVICES_EXPENDITURE  OTHER_EXPENDITURE  ...  G08_HI_A_READING  \
0                     7.350360e+05      429950.933824  ...        254.845528   
1                     3.509020e+05      429950.933824  ...        254.845528   
2                     1.007732e+06      429950.933824  ...        254.845528   
3                     4.834880e+05      429950.933824  ...        254.845528   
4                     8.520926e+06      429950.933824  ...        254.845528   
...                            ...                ...  ...               ...   
1710                  2.682587e+06      429950.933824  ...        247.000000   
1711                  2.682587e+06      429950.933824  ...        248.000000   
1712                  2.682587e+06      429950.933824  ...        254.845528   
1713                  2.682587e+06      429950.933824  ...        251.000000   
1714                  2.682587e+06      429950.933824  ...        254.000000   

      G08_HI_A_MATHEMATICS  G08_AS_A_READING  G08_AS_A_MATHEMATICS  \
0               269.995968        279.803922            306.898089   
1               269.995968        279.803922            306.898089   
2               269.995968        279.803922            306.898089   
3               269.995968        279.803922            306.898089   
4               269.995968        279.803922            306.898089   
...                    ...               ...                   ...   
1710            278.000000        286.000000            315.000000   
1711            267.000000        285.000000            315.000000   
1712            269.995968        279.803922            306.898089   
1713            273.000000        277.000000            294.000000   
1714            274.000000        279.803922            306.898089   

      G08_AM_A_READING  G08_AM_A_MATHEMATICS  G08_HP_A_READING  \
0           246.688525            261.016667        248.285714   
1           246.688525            261.016667        248.285714   
2           246.688525            261.016667        248.285714   
3           246.688525            261.016667        248.285714   
4           246.688525            261.016667        248.285714   
...                ...                   ...               ...   
1710        246.688525  

In [15]:
data

PRIMARY_KEY          STATE  YEAR  ENROLL  TOTAL_REVENUE  \
0           1992_ALABAMA        ALABAMA  1992     NaN      2678885.0   
1            1992_ALASKA         ALASKA  1992     NaN      1049591.0   
2           1992_ARIZONA        ARIZONA  1992     NaN      3258079.0   
3          1992_ARKANSAS       ARKANSAS  1992     NaN      1711959.0   
4        1992_CALIFORNIA     CALIFORNIA  1992     NaN     26260025.0   
...                  ...            ...   ...     ...            ...   
1710       2019_VIRGINIA       VIRGINIA  2019     NaN            NaN   
1711     2019_WASHINGTON     WASHINGTON  2019     NaN            NaN   
1712  2019_WEST_VIRGINIA  WEST_VIRGINIA  2019     NaN            NaN   
1713      2019_WISCONSIN      WISCONSIN  2019     NaN            NaN   
1714        2019_WYOMING        WYOMING  2019     NaN            NaN   

      FEDERAL_REVENUE  STATE_REVENUE  LOCAL_REVENUE  TOTAL_EXPENDITURE  \
0            304177.0      1659028.0       715680.0          2653798.0   
1            106780.0       720711.0       222100.0           972488.0   
2            297888.0      1369815.0      1590376.0          3401580.0   
3            178571.0       958785.0       574603.0          1743022.0   
4           2072470.0     16546514.0      7641041.0         27138832.0   
...               ...            ...            ...                ...   
1710              NaN            NaN            NaN                NaN   
1711              NaN            NaN            NaN                NaN   
1712              NaN            NaN            NaN                NaN   
1713              NaN            NaN            NaN                NaN   
1714              NaN            NaN            NaN                NaN   

      INSTRUCTION_EXPENDITURE  ...  G08_HI_A_READING  G08_HI_A_MATHEMATICS  \
0                   1481703.0  ...               NaN                   NaN   
1                    498362.0  ...               NaN                   NaN   
2                   1435908.0  ...               NaN                   NaN   
3                    964323.0  ...               NaN                   NaN   
4                  14358922.0  ...               NaN                   NaN   
...                       ...  ...               ...                   ...   
1710                      NaN  ...             247.0                 278.0   
1711                      NaN  ...             248.0                 267.0   
1712                      NaN  ...               NaN                   NaN   
1713                      NaN  ...             251.0                 273.0   
1714                      NaN  ...             254.0                 274.0   

      G08_AS_A_READING  G08_AS_A_MATHEMATICS  G08_AM_A_READING  \
0                  NaN                   NaN               NaN   
1                  NaN                   NaN               NaN   
2                  NaN                   NaN               NaN   
3                  NaN                   NaN               NaN   
4                  NaN                   NaN               NaN   
...                ...                   ...               ...   
1710             286.0                 315.0               NaN   
1711             285.0                 315.0             237.0   
1712               NaN                   NaN               NaN   
1713             277.0                 294.0             253.0   
1714               NaN                   NaN             243.0   

      G08_AM_A_MATHEMATICS  G08_HP_A_READING  G08_HP_A_MATHEMATICS  \
0                      NaN               NaN                   NaN   
1                      NaN               NaN                   NaN   
2                      NaN               NaN                   NaN   
3                      NaN               NaN                   NaN   
4                      NaN               NaN                   NaN   
...                    ...               ...                   ...   
1710                   NaN             

In [16]:
for c in liste:
    data[c]=df[c]

In [17]:
data.head(20)

PRIMARY_KEY                 STATE  YEAR         ENROLL  \
0                1992_ALABAMA               ALABAMA  1992  917541.566176   
1                 1992_ALASKA                ALASKA  1992  917541.566176   
2                1992_ARIZONA               ARIZONA  1992  917541.566176   
3               1992_ARKANSAS              ARKANSAS  1992  917541.566176   
4             1992_CALIFORNIA            CALIFORNIA  1992  917541.566176   
5               1992_COLORADO              COLORADO  1992  917541.566176   
6            1992_CONNECTICUT           CONNECTICUT  1992  917541.566176   
7               1992_DELAWARE              DELAWARE  1992  917541.566176   
8   1992_DISTRICT_OF_COLUMBIA  DISTRICT_OF_COLUMBIA  1992  917541.566176   
9                1992_FLORIDA               FLORIDA  1992  917541.566176   
10               1992_GEORGIA               GEORGIA  1992  917541.566176   
11                1992_HAWAII                HAWAII  1992  917541.566176   
12                 1992_IDAHO                 IDAHO  1992  917541.566176   
13              1992_ILLINOIS              ILLINOIS  1992  917541.566176   
14               1992_INDIANA               INDIANA  1992  917541.566176   
15                  1992_IOWA                  IOWA  1992  917541.566176   
16                1992_KANSAS                KANSAS  1992  917541.566176   
17              1992_KENTUCKY              KENTUCKY  1992  917541.566176   
18             1992_LOUISIANA             LOUISIANA  1992  917541.566176   
19                 1992_MAINE                 MAINE  1992  917541.566176   

    TOTAL_REVENUE  FEDERAL_REVENUE  STATE_REVENUE  LOCAL_REVENUE  \
0       2678885.0         304177.0      1659028.0       715680.0   
1       1049591.0         106780.0       720711.0       222100.0   
2       3258079.0         297888.0      1369815.0      1590376.0   
3       1711959.0         178571.0       958785.0       574603.0   
4      26260025.0        2072470.0     16546514.0      7641041.0   
5       3185173.0         163253.0      1307986.0      1713934.0   
6       3834302.0         143542.0      1342539.0      2348221.0   
7        645233.0          45945.0       420942.0       178346.0   
8        709480.0          64749.0            0.0       644731.0   
9      11506299.0         788420.0      5683949.0      5033930.0   
10      5536901.0         398701.0      2798674.0      2339526.0   
11       996809.0          71273.0       903443.0        22093.0   
12       859329.0          69138.0       531912.0       258279.0   
13      9748650.0         597077.0      3073178.0      6078395.0   
14      5060274.0         257044.0      2578149.0      2225081.0   
15      2663934.0         125665.0      1176205.0      1362064.0   
16      2245348.0         121625.0       903199.0      1220524.0   
17      2642902.0         289902.0      1643109.0       709891.0   
18      3341253.0         367822.0      1848734.0      1124697.0   
19      1206819.0          73160.0       511654.0       622005.0   

    TOTAL_EXPENDITURE  INSTRUCTION_EXPENDITURE  ...  G08_HI_A_READING  \
0           2653798.0                1481703.0  ...        254.845528   
1            972488.0                 498362.0  ...        254.845528   
2           3401580.0                1435908.0  ...        254.845528   
3           1743022.0                 964323.0  ...        254.845528   
4          27138832.0               14358922.0  ...        254.845528   
5           3264826.0                1642466.0  ...        254.845528   
6           3721338.0                2148041.0  ...        254.845528   
7            638784.0                 372722.0  ...        254.845528   
8            742893.0                 329160.0  ...        254.845528   
9          11305642.0                5166374.0  ...        254.845528   
10          5535942.0                3043984.0  ...        254.845528   
11          1040121.0                 536115.0  ...        254.845528   
12           886161.0                 47350

% 90 ı "nan" olan bir variable ı sample almadan veya katagoriye ayırmadan doldurmanın bir anlamı yok

In [18]:
data_2 = pd.read_csv("/Users/aytackolukisa/Desktop/Jupiter_Dosyaları/states_all_extended.csv")

In [19]:
data_2.head(5)

PRIMARY_KEY       STATE  YEAR  ENROLL  TOTAL_REVENUE  FEDERAL_REVENUE  \
0     1992_ALABAMA     ALABAMA  1992     NaN      2678885.0         304177.0   
1      1992_ALASKA      ALASKA  1992     NaN      1049591.0         106780.0   
2     1992_ARIZONA     ARIZONA  1992     NaN      3258079.0         297888.0   
3    1992_ARKANSAS    ARKANSAS  1992     NaN      1711959.0         178571.0   
4  1992_CALIFORNIA  CALIFORNIA  1992     NaN     26260025.0        2072470.0   

   STATE_REVENUE  LOCAL_REVENUE  TOTAL_EXPENDITURE  INSTRUCTION_EXPENDITURE  \
0      1659028.0       715680.0          2653798.0                1481703.0   
1       720711.0       222100.0           972488.0                 498362.0   
2      1369815.0      1590376.0          3401580.0                1435908.0   
3       958785.0       574603.0          1743022.0                 964323.0   
4     16546514.0      7641041.0         27138832.0               14358922.0   

   ...  G08_HI_A_READING  G08_HI_A_MATHEMATICS  G08_AS_A_READING  \
0  ...               NaN                   NaN               NaN   
1  ...               NaN                   NaN               NaN   
2  ...               NaN                   NaN               NaN   
3  ...               NaN                   NaN               NaN   
4  ...               NaN                   NaN               NaN   

   G08_AS_A_MATHEMATICS  G08_AM_A_READING  G08_AM_A_MATHEMATICS  \
0                   NaN               NaN                   NaN   
1                   NaN               NaN                   NaN   
2                   NaN               NaN                   NaN   
3                   NaN               NaN                   NaN   
4                   NaN               NaN                   NaN   

   G08_HP_A_READING  G08_HP_A_MATHEMATICS  G08_TR_A_READING  \
0               NaN                   NaN               NaN   
1               NaN                   NaN               NaN   
2               NaN                   NaN               NaN   
3               NaN                   NaN               NaN   
4               NaN                   NaN               NaN   

   G08_TR_A_MATHEMATICS  
0                   NaN  
1                   NaN  
2                   NaN  
3                   NaN  
4                   NaN  

[5 rows x 266 columns]

In [20]:
a=data_2.groupby('YEAR')['ENROLL'].mean()

In [21]:
a

YEAR
1986              NaN
1987              NaN
1988              NaN
1989              NaN
1990              NaN
1991              NaN
1992              NaN
1993    802054.333333
1994    801383.960784
1995    858973.039216
1996    873930.607843
1997    888379.745098
1998    897509.058824
1999    905101.470588
2000    910446.705882
2001    916296.274510
2002    923818.196078
2003    933396.862745
2004    938327.117647
2005    942259.960784
2006    947078.980392
2007    949109.254902
2008    949047.392157
2009    946249.313725
2010    946396.254902
2011    946587.745098
2012    945379.392157
2013    947012.176471
2014    948607.529412
2015    951263.450980
2016    952388.764706
2017              NaN
2019              NaN
Name: ENROLL, dtype: float64

In [22]:
a.values[0]

nan

In [23]:
len(a)

33

In [40]:
def ortalama (variable):
    a=df1.groupby('YEAR')[variable].mean()
    liste=np.arange(len(a))
    for i in liste :
        data_2[variable].fillna(a.values[i], inplace = True)

In [41]:
a

YEAR
1986              NaN
1987              NaN
1988              NaN
1989              NaN
1990              NaN
1991              NaN
1992              NaN
1993    802054.333333
1994    801383.960784
1995    858973.039216
1996    873930.607843
1997    888379.745098
1998    897509.058824
1999    905101.470588
2000    910446.705882
2001    916296.274510
2002    923818.196078
2003    933396.862745
2004    938327.117647
2005    942259.960784
2006    947078.980392
2007    949109.254902
2008    949047.392157
2009    946249.313725
2010    946396.254902
2011    946587.745098
2012    945379.392157
2013    947012.176471
2014    948607.529412
2015    951263.450980
2016    952388.764706
2017              NaN
2019              NaN
Name: ENROLL, dtype: float64

In [52]:
a.values[0]

nan

In [42]:
df1 = data_2.drop(["PRIMARY_KEY","STATE"],axis=1)

In [43]:
liste_3 =df1.columns

In [44]:
for x in liste_3:
    ortalama(x)

In [45]:
df1

YEAR         ENROLL  TOTAL_REVENUE  FEDERAL_REVENUE  STATE_REVENUE  \
0     1992  802054.333333   2.678885e+06     3.041770e+05   1.659028e+06   
1     1992  802054.333333   1.049591e+06     1.067800e+05   7.207110e+05   
2     1992  802054.333333   3.258079e+06     2.978880e+05   1.369815e+06   
3     1992  802054.333333   1.711959e+06     1.785710e+05   9.587850e+05   
4     1992  802054.333333   2.626002e+07     2.072470e+06   1.654651e+07   
...    ...            ...            ...              ...            ...   
1710  2019  802054.333333   4.572764e+06     2.970296e+05   2.088314e+06   
1711  2019  802054.333333   4.572764e+06     2.970296e+05   2.088314e+06   
1712  2019  802054.333333   4.572764e+06     2.970296e+05   2.088314e+06   
1713  2019  802054.333333   4.572764e+06     2.970296e+05   2.088314e+06   
1714  2019  802054.333333   4.572764e+06     2.970296e+05   2.088314e+06   

      LOCAL_REVENUE  TOTAL_EXPENDITURE  INSTRUCTION_EXPENDITURE  \
0      7.156800e+05       2.653798e+06             1.481703e+06   
1      2.221000e+05       9.724880e+05             4.983620e+05   
2      1.590376e+06       3.401580e+06             1.435908e+06   
3      5.746030e+05       1.743022e+06             9.643230e+05   
4      7.641041e+06       2.713883e+07             1.435892e+07   
...             ...                ...                      ...   
1710   2.187420e+06       4.653591e+06             2.454000e+06   
1711   2.187420e+06       4.653591e+06             2.454000e+06   
1712   2.187420e+06       4.653591e+06             2.454000e+06   
1713   2.187420e+06       4.653591e+06             2.454000e+06   
1714   2.187420e+06       4.653591e+06             2.454000e+06   

      SUPPORT_SERVICES_EXPENDITURE  OTHER_EXPENDITURE  ...  G08_HI_A_READING  \
0                     7.350360e+05      248234.784314  ...        253.708333   
1                     3.509020e+05      248234.784314  ...        253.708333   
2                     1.007732e+06      248234.784314  ...        253.708333   
3                     4.834880e+05      248234.784314  ...        253.708333   
4                     8.520926e+06      248234.784314  ...        253.708333   
...                            ...                ...  ...               ...   
1710                  1.370149e+06      248234.784314  ...        247.000000   
1711                  1.370149e+06      248234.784314  ...        248.000000   
1712                  1.370149e+06      248234.784314  ...        253.708333   
1713                  1.370149e+06      248234.784314  ...        251.000000   
1714                  1.370149e+06      248234.784314  ...        254.000000   

      G08_HI_A_MATHEMATICS  G08_AS_A_READING  G08_AS_A_MATHEMATICS  \
0               269.897959        276.354839            304.032258   
1               269.897959        276.354839            304.032258   
2               269.897959        276.354839            304.032258   
3               269.897959        276.354839            304.032258   
4               269.897959        276.354839            304.032258   
...                    ...               ...                   ...   
1710            278.000000        286.000000            315.000000   
1711            267.000000        285.000000            315.000000   
1712            269.897959        276.354839            304.032258   
1713            273.000000        277.000000            294.000000   
1714            274.000000        276.354839            304.032258   

      G08_AM_A_READING  G08_AM_A_MATHEMATICS  G08_HP_A_READING  \
0           248.307692            260.461538             249.0   
1           248.307692            260.461538             249.0   
2           248.307692            260.461538             249.0   
3           248.307692            260.461538             249.0   
4           248.307692            260.461538             249.0   
...                ...                   ...               ...   
1710        248.307692  

In [46]:
liste = df1.columns

In [47]:
for c in liste:
    data_2[c]=df1[c]

In [48]:
data_2

PRIMARY_KEY          STATE  YEAR         ENROLL  TOTAL_REVENUE  \
0           1992_ALABAMA        ALABAMA  1992  802054.333333   2.678885e+06   
1            1992_ALASKA         ALASKA  1992  802054.333333   1.049591e+06   
2           1992_ARIZONA        ARIZONA  1992  802054.333333   3.258079e+06   
3          1992_ARKANSAS       ARKANSAS  1992  802054.333333   1.711959e+06   
4        1992_CALIFORNIA     CALIFORNIA  1992  802054.333333   2.626002e+07   
...                  ...            ...   ...            ...            ...   
1710       2019_VIRGINIA       VIRGINIA  2019  802054.333333   4.572764e+06   
1711     2019_WASHINGTON     WASHINGTON  2019  802054.333333   4.572764e+06   
1712  2019_WEST_VIRGINIA  WEST_VIRGINIA  2019  802054.333333   4.572764e+06   
1713      2019_WISCONSIN      WISCONSIN  2019  802054.333333   4.572764e+06   
1714        2019_WYOMING        WYOMING  2019  802054.333333   4.572764e+06   

      FEDERAL_REVENUE  STATE_REVENUE  LOCAL_REVENUE  TOTAL_EXPENDITURE  \
0        3.041770e+05   1.659028e+06   7.156800e+05       2.653798e+06   
1        1.067800e+05   7.207110e+05   2.221000e+05       9.724880e+05   
2        2.978880e+05   1.369815e+06   1.590376e+06       3.401580e+06   
3        1.785710e+05   9.587850e+05   5.746030e+05       1.743022e+06   
4        2.072470e+06   1.654651e+07   7.641041e+06       2.713883e+07   
...               ...            ...            ...                ...   
1710     2.970296e+05   2.088314e+06   2.187420e+06       4.653591e+06   
1711     2.970296e+05   2.088314e+06   2.187420e+06       4.653591e+06   
1712     2.970296e+05   2.088314e+06   2.187420e+06       4.653591e+06   
1713     2.970296e+05   2.088314e+06   2.187420e+06       4.653591e+06   
1714     2.970296e+05   2.088314e+06   2.187420e+06       4.653591e+06   

      INSTRUCTION_EXPENDITURE  ...  G08_HI_A_READING  G08_HI_A_MATHEMATICS  \
0                1.481703e+06  ...        253.708333            269.897959   
1                4.983620e+05  ...        253.708333            269.897959   
2                1.435908e+06  ...        253.708333            269.897959   
3                9.643230e+05  ...        253.708333            269.897959   
4                1.435892e+07  ...        253.708333            269.897959   
...                       ...  ...               ...                   ...   
1710             2.454000e+06  ...        247.000000            278.000000   
1711             2.454000e+06  ...        248.000000            267.000000   
1712             2.454000e+06  ...        253.708333            269.897959   
1713             2.454000e+06  ...        251.000000            273.000000   
1714             2.454000e+06  ...        254.000000            274.000000   

      G08_AS_A_READING  G08_AS_A_MATHEMATICS  G08_AM_A_READING  \
0           276.354839            304.032258        248.307692   
1           276.354839            304.032258        248.307692   
2           276.354839            304.032258        248.307692   
3           276.354839            304.032258        248.307692   
4           276.354839            304.032258        248.307692   
...                ...                   ...               ...   
1710        286.000000            315.000000        248.307692   
1711        285.000000            315.000000        237.000000   
1712        276.354839            304.032258        248.307692   
1713        277.000000            294.000000        253.000000   
1714        276.354839            304.032258        243.000000   

      G08_AM_A_MATHEMATICS  G08_HP_A_READING  G08_HP_A_MATHEMATICS  \
0               260.461538             249.0                 266.0   
1               260.461538             249.0                 266.0   
2               260.461538             249.0                 266.0   
3               260.461538             249.0                 266.0   
4               260.461538             249.0                 266.0   
...                    ..

In [49]:
data

PRIMARY_KEY          STATE  YEAR         ENROLL  TOTAL_REVENUE  \
0           1992_ALABAMA        ALABAMA  1992  917541.566176   2.678885e+06   
1            1992_ALASKA         ALASKA  1992  917541.566176   1.049591e+06   
2           1992_ARIZONA        ARIZONA  1992  917541.566176   3.258079e+06   
3          1992_ARKANSAS       ARKANSAS  1992  917541.566176   1.711959e+06   
4        1992_CALIFORNIA     CALIFORNIA  1992  917541.566176   2.626002e+07   
...                  ...            ...   ...            ...            ...   
1710       2019_VIRGINIA       VIRGINIA  2019  917541.566176   9.102045e+06   
1711     2019_WASHINGTON     WASHINGTON  2019  917541.566176   9.102045e+06   
1712  2019_WEST_VIRGINIA  WEST_VIRGINIA  2019  917541.566176   9.102045e+06   
1713      2019_WISCONSIN      WISCONSIN  2019  917541.566176   9.102045e+06   
1714        2019_WYOMING        WYOMING  2019  917541.566176   9.102045e+06   

      FEDERAL_REVENUE  STATE_REVENUE  LOCAL_REVENUE  TOTAL_EXPENDITURE  \
0        3.041770e+05   1.659028e+06   7.156800e+05       2.653798e+06   
1        1.067800e+05   7.207110e+05   2.221000e+05       9.724880e+05   
2        2.978880e+05   1.369815e+06   1.590376e+06       3.401580e+06   
3        1.785710e+05   9.587850e+05   5.746030e+05       1.743022e+06   
4        2.072470e+06   1.654651e+07   7.641041e+06       2.713883e+07   
...               ...            ...            ...                ...   
1710     7.677799e+05   4.223743e+06   4.110522e+06       9.206242e+06   
1711     7.677799e+05   4.223743e+06   4.110522e+06       9.206242e+06   
1712     7.677799e+05   4.223743e+06   4.110522e+06       9.206242e+06   
1713     7.677799e+05   4.223743e+06   4.110522e+06       9.206242e+06   
1714     7.677799e+05   4.223743e+06   4.110522e+06       9.206242e+06   

      INSTRUCTION_EXPENDITURE  ...  G08_HI_A_READING  G08_HI_A_MATHEMATICS  \
0                1.481703e+06  ...        254.845528            269.995968   
1                4.983620e+05  ...        254.845528            269.995968   
2                1.435908e+06  ...        254.845528            269.995968   
3                9.643230e+05  ...        254.845528            269.995968   
4                1.435892e+07  ...        254.845528            269.995968   
...                       ...  ...               ...                   ...   
1710             4.768010e+06  ...        247.000000            278.000000   
1711             4.768010e+06  ...        248.000000            267.000000   
1712             4.768010e+06  ...        254.845528            269.995968   
1713             4.768010e+06  ...        251.000000            273.000000   
1714             4.768010e+06  ...        254.000000            274.000000   

      G08_AS_A_READING  G08_AS_A_MATHEMATICS  G08_AM_A_READING  \
0           279.803922            306.898089        246.688525   
1           279.803922            306.898089        246.688525   
2           279.803922            306.898089        246.688525   
3           279.803922            306.898089        246.688525   
4           279.803922            306.898089        246.688525   
...                ...                   ...               ...   
1710        286.000000            315.000000        246.688525   
1711        285.000000            315.000000        237.000000   
1712        279.803922            306.898089        246.688525   
1713        277.000000            294.000000        253.000000   
1714        279.803922            306.898089        243.000000   

      G08_AM_A_MATHEMATICS  G08_HP_A_READING  G08_HP_A_MATHEMATICS  \
0               261.016667        248.285714            264.846154   
1               261.016667        248.285714            264.846154   
2               261.016667        248.285714            264.846154   
3               261.016667        248.285714            264.846154   
4               261.016667        248.285714            264.846154   
...                    ..

In [50]:
data_e = pd.read_csv("/Users/aytackolukisa/Desktop/Jupiter_Dosyaları/states_all_extended.csv")

In [58]:
def enterpolasyon (variabel):
    liste = np.arange(len (data_e1))
    for i in liste:
        if data_e1[variabel][i] ==  "nan":
            data_e1[variabel][i] == ( data_e1[variabel][i-1] + data_e1[variabel][i+1])/2
        

In [59]:
data_e1 = data_e.drop(["PRIMARY_KEY","STATE"],axis=1)

In [60]:
liste4 = data_e1.columns

In [61]:
for c in liste4:
    enterpolasyon(c)
    

In [63]:
liste5 = data_e1.columns

In [66]:
for x in liste4:
    data_e[x]=data_e1[x]
    

In [67]:
data_e

PRIMARY_KEY          STATE  YEAR  ENROLL  TOTAL_REVENUE  \
0           1992_ALABAMA        ALABAMA  1992     NaN      2678885.0   
1            1992_ALASKA         ALASKA  1992     NaN      1049591.0   
2           1992_ARIZONA        ARIZONA  1992     NaN      3258079.0   
3          1992_ARKANSAS       ARKANSAS  1992     NaN      1711959.0   
4        1992_CALIFORNIA     CALIFORNIA  1992     NaN     26260025.0   
...                  ...            ...   ...     ...            ...   
1710       2019_VIRGINIA       VIRGINIA  2019     NaN            NaN   
1711     2019_WASHINGTON     WASHINGTON  2019     NaN            NaN   
1712  2019_WEST_VIRGINIA  WEST_VIRGINIA  2019     NaN            NaN   
1713      2019_WISCONSIN      WISCONSIN  2019     NaN            NaN   
1714        2019_WYOMING        WYOMING  2019     NaN            NaN   

      FEDERAL_REVENUE  STATE_REVENUE  LOCAL_REVENUE  TOTAL_EXPENDITURE  \
0            304177.0      1659028.0       715680.0          2653798.0   
1            106780.0       720711.0       222100.0           972488.0   
2            297888.0      1369815.0      1590376.0          3401580.0   
3            178571.0       958785.0       574603.0          1743022.0   
4           2072470.0     16546514.0      7641041.0         27138832.0   
...               ...            ...            ...                ...   
1710              NaN            NaN            NaN                NaN   
1711              NaN            NaN            NaN                NaN   
1712              NaN            NaN            NaN                NaN   
1713              NaN            NaN            NaN                NaN   
1714              NaN            NaN            NaN                NaN   

      INSTRUCTION_EXPENDITURE  ...  G08_HI_A_READING  G08_HI_A_MATHEMATICS  \
0                   1481703.0  ...               NaN                   NaN   
1                    498362.0  ...               NaN                   NaN   
2                   1435908.0  ...               NaN                   NaN   
3                    964323.0  ...               NaN                   NaN   
4                  14358922.0  ...               NaN                   NaN   
...                       ...  ...               ...                   ...   
1710                      NaN  ...             247.0                 278.0   
1711                      NaN  ...             248.0                 267.0   
1712                      NaN  ...               NaN                   NaN   
1713                      NaN  ...             251.0                 273.0   
1714                      NaN  ...             254.0                 274.0   

      G08_AS_A_READING  G08_AS_A_MATHEMATICS  G08_AM_A_READING  \
0                  NaN                   NaN               NaN   
1                  NaN                   NaN               NaN   
2                  NaN                   NaN               NaN   
3                  NaN                   NaN               NaN   
4                  NaN                   NaN               NaN   
...                ...                   ...               ...   
1710             286.0                 315.0               NaN   
1711             285.0                 315.0             237.0   
1712               NaN                   NaN               NaN   
1713             277.0                 294.0             253.0   
1714               NaN                   NaN             243.0   

      G08_AM_A_MATHEMATICS  G08_HP_A_READING  G08_HP_A_MATHEMATICS  \
0                      NaN               NaN                   NaN   
1                      NaN               NaN                   NaN   
2                      NaN               NaN                   NaN   
3                      NaN               NaN                   NaN   
4                      NaN               NaN                   NaN   
...                    ...               ...                   ...   
1710                   NaN             